pkg/router/controller/status_test.go

```diff
@@ -373,11 +346,134 @@ func TestStatusRecordRejectionConflict(t *testing.T) {
 	if condition.LastTransitionTime == nil || *condition.LastTransitionTime != now || condition.Status != kapi.ConditionFalse || condition.Reason != "Failed" || condition.Message != "generic error" {
 		t.Fatalf("unexpected condition: %#v", condition)
 	}
-	if v, ok := admitter.expected.Peek(types.UID("uid1")); !ok || !reflect.DeepEqual(v, time.Time{}) {
+	if v, ok := admitter.expected.Peek(types.UID("uid1")); ok {
 		t.Fatalf("expected empty time: %#v", v)
 	}
 }
 
+func TestStatusFightBetweenReplicas(t *testing.T) {
+	p := &fakePlugin{}
+
+	// the initial pre-population
+	now1 := unversioned.Now()
+	nowFn = func() unversioned.Time { return now1 }
+	c1 := testclient.NewSimpleFake(&routeapi.Route{})
+	admitter1 := NewStatusAdmitter(p, c1, "test")
+	err := admitter1.HandleRoute(watch.Added, &routeapi.Route{
+		ObjectMeta: kapi.ObjectMeta{Name: "route1", Namespace: "default", UID: types.UID("uid1")},
+		Spec:       routeapi.RouteSpec{Host: "route1.test.local"},
+		Status:     routeapi.RouteStatus{},
+	})
+
+	outObj1 := checkResult(t, err, c1, admitter1, "route1.test.local", now1, &now1.Time, 0, 0)
+
+	// the new deployment's replica
+	now2 := unversioned.Time{Time: now1.Time.Add(time.Minute)}
+	nowFn = func() unversioned.Time { return now2 }
+	c2 := testclient.NewSimpleFake(&routeapi.Route{})
+	admitter2 := NewStatusAdmitter(p, c2, "test")
+	outObj1.Spec.Host = "route1.test-new.local"
+	err = admitter2.HandleRoute(watch.Added, outObj1)
+
+	outObj2 := checkResult(t, err, c2, admitter2, "route1.test-new.local", now2, &now2.Time, 0, 0)
+
+	now3 := unversioned.Time{Time: now1.Time.Add(time.Minute)}
+	nowFn = func() unversioned.Time { return now3 }
+	outObj2.Spec.Host = "route1.test.local"
+	err = admitter1.HandleRoute(watch.Modified, outObj2)
+
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	// expect the last HandleRoute not to have performed any actions
+	if len(c1.Actions()) != 1 {
+		t.Fatalf("unexpected actions: %#v", c1.Actions())
+	}
+
+}
+
+func TestStatusFightBetweenRouters(t *testing.T) {
```

Comment:
I think we *might* want to run a longer sequence of random fighting (increment one or the other randomly and invoke handle route in a for loop), then simulate a jump forward in the clock (set now past the content interval) and verify only one ends up winning.

Would do that as a separate test:

i.e.:

```
for i ... 100
  if rand.Bool()
    mutateA
  else
    mutateB
  verify postcondition still holds

advance clock > contention interval

for 1..100
```

Might even want to simulate 3.  It should be fast to run, but I really want to ensure we have a realistic "fight".